pkg/tcpip/transport/tcp/accept.go

```diff
@@ -202,6 +202,7 @@ func (l *listenContext) createConnectingEndpoint(s *segment, rcvdSynOpts header.
 	n.boundNICID = s.pkt.NICID
 	n.route = route
 	n.effectiveNetProtos = []tcpip.NetworkProtocolNumber{s.pkt.NetworkProtocolNumber}
+	n.userCookie = s.pkt.UserCookie
```

Comment:
I'm passing the user cookie from the packet (if specified) to the TCP endpoint/connection. This allows marking incoming packets with user supplied metadata (eg. lower layer information).